cherrypy/test/test_core.py

```diff
@@ -824,3 +824,19 @@ def test_bind_ephemeral_port(self):
         assert port > 0
         cherrypy.engine.stop()
         assert cherrypy.server.bind_addr == cherrypy.server.bound_addr
+
+
+class TestScriptName(helper.CPWebCase):
+    @staticmethod
+    def setup_server():
+        class Root:
+
+            @cherrypy.expose
+            def bar(self):
+                return 'foobar'
+
+        cherrypy.tree.mount(Root(), script_name='foo')
```

Comment:
Please also test `script_name=None`, `script_name=''` and `script_name='/'`.

It looks like it changes the behavior described in the docstring of `Tree.mount()`:
>         script_name
>             ... 
>
>             It MUST NOT end in a slash. If the script_name refers to the
>             root of the URI, it MUST be an empty string (not "/").

Reply:
Normally, I'd be reluctant to ask someone to add tests for behavior outside the scope of the behavior they're trying to support, but in this case, I'm pretty sure there's a regression for untested behavior here, so adding those tests is important. If your `git` foo is good enough, consider adding the tests against master first, so you can see the expected passing behavior on the examples that webknjaz suggested, but also the failing behavior on `script_name='foo'`.